### PR TITLE
Certificates in templates

### PR DESCRIPTION
**Pull request by maraino**
### Description

This PR adds fixed keys and methods to add certificates and certificate chains to the templates. If a provisioner adds those methods they will be accessible from the template like `{{ .AuthorizationCrt }}` or `{{ .AuthorizationChain }}`.

### CODE REVIEW

**Comment by tashian** [comment]
Hey @maraino I just started putting some docs together for this.
Where can I find the list of properties that are available in the template, for .Crt and .Chain?

File: sshutil/templates.go [hunk 1]
```diff
@@ -134,6 +136,18 @@ func (t TemplateData) SetUserData(v interface{}) {
 	t.SetInsecure(UserKey, v)
 }
 
+// SetCertificate sets a the given certificate in the template. This certificate
```
Comment (dopey) [inline sshutil/templates.go:139]: sets the given - remove `a`

**Review by dopey — APPROVED** [review]
lgtm. left one comment.

File: sshutil/templates.go (hunk 1 above)
Comment (maraino) [reply sshutil/templates.go]: Fixed with b9132b1

**Pull request merged**